code/modules/projectiles/guns/projectile.dm

```diff
@@ -96,9 +97,9 @@
 /obj/item/weapon/gun/projectile/proc/load_ammo(var/obj/item/A, mob/user)
 	if(istype(A, /obj/item/ammo_magazine))
 		var/obj/item/ammo_magazine/AM = A
-		if(!(load_method & AM.mag_type) || caliber != AM.caliber)
-			return //incompatible
-
+		if(!(load_method & AM.mag_type) || caliber != AM.caliber || allowed_magazines && !is_type_in_list(A, allowed_magazines))

```

Comment:
I wonder if this is better written as:
`if(!(load_method & AM.mag_type) || caliber != AM.caliber || (allowed_magazines && !is_type_in_list(A, allowed_magazines)))`?

As we're dealing with 3 unique failure scenarios, but one of them includes two conditions, it'd be neat if they're cordoned off with parentheses.